### ISSUE TITLE
No overload variant of "dict"/"list" matches argument types

### ISSUE THREAD

**Issue by gordianknot1981** [JukkaL/mypy#566]
```python
from typing import Dict, List
a = 'hello'
d = Dict[str, int]({'a':1})
d = {'a':1}

d2 = Dict[str,int]({'b':1})
d2 = Dict[str, int](dict(a=1))  #No overload variant of "dict" matches argument types
d2 = dict()
d2 = dict(a=1)                #No overload variant of "dict" matches argument types

l = List[int]([])
l = List[int](list(1,2,3))     #No overload variant of "list" matches argument types
l = list()
l = list(1,2,3)                   #No overload variant of "list" matches argument types
```

is that by design?

**Comment by gvanrossum** [comment]
I think mypy doesn't know that you can call `dict()` with keyword args.

The last two are correct warnings: `list(1, 2, 3)` is wrong, should be `list((1, 2, 3))` and then mypy doesn't complain.

**Comment by JukkaL** [comment]
`dict(x=...)` is not supported right now, but it's a high-priority issue: https://github.com/JukkaL/mypy/issues/239

As Guido pointed out, rejecting `list(1, 2, 3)` is actually a feature :-)
